nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -0,0 +1,128 @@
+define([
+    'uuid',
+    'common/appUtils',
+    'common/utils',
+    'common/runtime',
+    'common/busEventManager',
+    'base/js/namespace'
+], (
+    Uuid,
+    AppUtils,
+    Utils,
+    Runtime,
+    BusEventManager,
+    Jupyter
+) => {
+    'use strict';
+    const CELL_TYPE = 'app-bulk-import';
+
+    class BulkImportCell {
+        constructor(cell, initialize) {
+            if (cell.cell_type !== 'code') {
+                throw new Error('Can only create Bulk Import Cells out of code cells!');
+            }
+            this.cell = cell;
+            this.kbaseNode = null;
+            this.runtime = Runtime.make();
+            this.busEventManager = BusEventManager.make({
+                bus: this.runtime.bus()
+            });
+            if (initialize) {
+                this.initialize();
```

Comment:
Does it make sense to have an error handling on this.initialize and this.setupcell?

Reply:
I figured that could get thrown outward and captured by the main controller. We can do error handling internally if that's preferred.

Reply:
Either way is fine as long as you don't think the errors will get swallowed.